python/grass/temporal/aggregation.py

```diff
@@ -316,6 +317,47 @@ def aggregate_by_topology(
         if "overlapped" in topo_list and granule.overlapped:
             for map_layer in granule.overlapped:
                 aggregation_list.append(map_layer.get_name())
+        if "related" in topo_list:
+            aggregation_weights = []
+            set_list = set()
+            try:
+                for map_layer in granule.overlaps:
+                    set_list.add(map_layer)
+            except:
+                pass
+            try:
+                for map_layer in granule.overlapped:
+                    set_list.add(map_layer)
+            except:
+                pass
+            try:
+                for map_layer in granule.contains:
+                    set_list.add(map_layer)
+            except:
+                pass
+            try:
+                for map_layer in granule.equal:
+                    set_list.add(map_layer)
+            except:
+                pass
+            try:
+                for map_layer in granule.during:
+                    set_list.add(map_layer)
+            except:
+                pass
+            if set_list:
+                for map_layer in set_list:
+                    aggregation_list.append(map_layer.get_name())
+                    t_granule_contained = map_layer.get_absolute_time()
+                    t_granule = granule.get_absolute_time()
+                    if None in t_granule_contained or None in t_granule:
+                        aggregation_weights.append(0)
+                    else:
+                        overlap_abs = min(t_granule[1], t_granule_contained[1]) - max(
+                            t_granule[0], t_granule_contained[0]
+                        )
```

Comment:
(In-code) comments for these lines would be appreciated. Easier to review, easier to maintain when the intention is documented.